packages/dht/src/connection/WebSocket/WebSocketConnector.ts

```diff
@@ -102,39 +101,39 @@ export class WebSocketConnector implements IWebSocketConnectorService {
                     this.attachHandshaker(connection)
                 }
             })
-            await this.webSocketServer.start(this.webSocketPort!, this.webSocketHost)
+            const port = await this.webSocketServer.start(this.portRange!, this.host)
+            this.selectedPort = port
+            this.connectivityChecker = new ConnectivityChecker(this.selectedPort!)
         }
     }
 
     public async checkConnectivity(reattempt = 0): Promise<ConnectivityResponse> {
+        const noServerConnectivityResponse: ConnectivityResponse = {
+            openInternet: false,
+            ip: '127.0.0.1',
+            natType: NatType.UNKNOWN
+        }
+        if (this.destroyed) {
+            return noServerConnectivityResponse
+        }
         try {
-            const noServerConnectivityResponse: ConnectivityResponse = {
-                openInternet: false,
-                ip: '127.0.0.1',
-                natType: NatType.UNKNOWN
-            }
-
             if (!this.webSocketServer) {
                 // If no websocket server, return openInternet: false
                 return noServerConnectivityResponse
             } else {
                 if (!this.entrypoints || this.entrypoints.length < 1) {
                     // return connectivity info given in config
-
                     const preconfiguredConnectivityResponse: ConnectivityResponse = {
                         openInternet: true,
-                        ip: this.webSocketHost!,
+                        ip: this.host!,
                         natType: NatType.OPEN_INTERNET,
-                        websocket: { ip: this.webSocketHost!, port: this.webSocketPort! }
+                        websocket: { ip: this.host!, port: this.selectedPort! }
                     }
                     return preconfiguredConnectivityResponse
                 } else {
-                    // Do real connectivity checking
-                    
+                    // Do real connectivity checking     
                     let response = noServerConnectivityResponse
-
-                    response = await this.connectivityChecker.sendConnectivityRequest(this.entrypoints[0])
-
+                    response = await this.connectivityChecker!.sendConnectivityRequest(sample(this.entrypoints)!)
```

Comment:
Do we need `let response = noServerConnectivityResponse`? We could just `return await this.connectivityChecker!.sendConnectivityRequest(sample(this.entrypoints)!)` in this block?